aiohttp/web_request.py

```diff
@@ -151,15 +181,60 @@ def secure(self):
         return self.url.scheme == 'https'
 
     @reify
+    def forwarded(self):
+        """ A frozendict containing parsed Forwarded header(s).
+
+        Makes an effort to parse Forwarded headers as specified by RFC 7239:
+
+        - It adds all parameters (by, for, host, proto) in the order it finds
+          them; starting at the topmost / first added 'Forwarded' header, at
+          the leftmost / first-added parwameter.
+        - It checks that the value has valid syntax in general as specified in
+          section 4: either a 'token' or a 'quoted-string'.
+        - It un-escapes found escape sequences.
+        - It does NOT validate 'by' and 'for' contents as specified in section
+          6.
+        - It does NOT validate 'host' contents (Host ABNF).
+        - It does NOT validate 'proto' contents for valid URI scheme names.
+
+        Returns a dict(by=tuple(...), for=tuple(...), host=tuple(...),
+        proto=tuple(...), ) 
+        """
+        params = {'by': [], 'for': [], 'host': [], 'proto': []}
+        if hdrs.FORWARDED in self._message.headers:
+            for forwarded_elm in self._message.headers.getall(hdrs.FORWARDED):
+                if len(forwarded_elm):
+                    forwarded_pairs = tuple(
+                        _FORWARDED_PAIR_RE.findall(pair)
+                        for pair in forwarded_elm.split(';'))
+                    for forwarded_pair in forwarded_pairs:
+                        if len(forwarded_pair) != 1:
+                            # non-compliant syntax, ignore
+                            continue
+                        param = forwarded_pair[0][0].lower()
+                        value = forwarded_pair[0][1]
+                        if len(value) and value[0] == '"':
```

Comment:
There's no need to check for a closing quote: the regex validates `quoted-string` so any value starting with `"` must also have a closing `"`.

A `token` may be empty (but note again that I don't check the syntax in section 6)